insights/parsers/dmesg.py

```diff
@@ -30,27 +30,75 @@
     True
     >>> dmesg.get('AGP')
     ['[    0.000000] AGP: Checking aperture...', '[    0.000000] AGP: No AGP bridge found']
-
 """
 
 from .. import LogFileOutput, parser
 
+import re
+
 
 @parser('dmesg')
 class DmesgLineList(LogFileOutput):
     """
     Class for reading output of ``dmesg`` using the LogFileOutput parser class.
     """
+    _line_re = re.compile(r'^(?:\[\s+(?P<timestamp>\d+\.\d+)\]\s+)?(?P<message>.*)$')
+
     def has_startswith(self, prefix):
         """
         Parameters:
-            prefix (str): The prefix of the line to look for.
+            prefix (str): The prefix of the line to look for.  Ignores any
+            timestamp before the message body.
 
         Returns:
             (bool): Does any line start with the given prefix?
 
-        Notes:
-            Does not ignore the time stamp of the line - if the dmesg output
-            contains timestamps then these are considered to start the line.
         """
-        return any(line.startswith(prefix) for line in self.lines)
+        def yield_messages(lines):
+            for line in lines:
+                match = self._line_re.search(line)
+                if match:
+                    yield match.group('message')
+                else:
+                    yield line
+        return any(msg.startswith(prefix) for msg in yield_messages(self.lines))
```

Comment:
OK, you made me look at this and I figured out a better way.

1. The regex always matches, since the message subgroup matches .* and everything else is optional.
2. Therefore we can always check that subgroup for the prefix.
3. Therefore we can remove the generator entirely and expand the test into the rather longwinded `self._line_re.search(line).group('message').startswith(prefix)`.

I've also tested for negative matches - i.e. that things that aren't in the log or aren't prefixes should not generate a match in `has_startswith()` - and fixed the documentation glitch.